### PR TITLE
Add --context for karmadactl init

### LINKED ISSUE #1744: Support specifying context when `init`
**What would you like to be added**:

When I install karmada into a cluster I can specify the kubeconfig. 

However, if there are multi contexts in the kubeconfig, I have no way to specify which host cluster to install karmada control plane. 
This is not friendly to local env, especially when using kind.

### PR DESCRIPTION
**Pull request by lonelyCZ**
Signed-off-by: lonelyCZ <531187475@qq.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #1744 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
`karmadactl`: Introduced `--context` flag to specify the context name to use.
```

### CODE REVIEW

File: pkg/karmadactl/cmdinit/utils/kubeclient.go [hunk 1]
```diff
@@ -9,18 +9,23 @@ import (
...
+
+	restConfig, err := clientConfig.ClientConfig()
 	if err != nil {
 		return nil, err
 	}
-	config.QPS = float32(5.000000)
-	config.Burst = 10
-	config.ContentType = "application/json"
-	config.AcceptContentTypes = "application/json"
-	config.UserAgent = rest.DefaultKubernetesUserAgent()
```
Comment (XiShanYongYe-Chang) [inline pkg/karmadactl/cmdinit/utils/kubeclient.go:21]: Will these deletions have any impact?

Comment (lonelyCZ) [reply pkg/karmadactl/cmdinit/utils/kubeclient.go:21]: There should no impacts. They all have default value.

https://github.com/karmada-io/karmada/blob/2d47c5b08a27132b2c2e2fa6ef080f54ee157c92/vendor/k8s.io/client-go/rest/config.go#L115-L121

https://github.com/karmada-io/karmada/blob/2d47c5b08a27132b2c2e2fa6ef080f54ee157c92/vendor/k8s.io/client-go/rest/config.go#L284-L294

Comment (XiShanYongYe-Chang) [reply pkg/karmadactl/cmdinit/utils/kubeclient.go:21]: got it.

Comment (RainbowMango) [reply pkg/karmadactl/cmdinit/utils/kubeclient.go:21]: What's the reason why we set them explicitly? @prodanlabs

**Comment by RainbowMango** [comment]
Generally looks good to me. Please @prodanlabs take a look.
(I updated the release notes to the PR description)

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**